### ISSUE TITLE
Pathfinding on chests

### ISSUE THREAD

**Issue by SaubereSache** [PrismarineJS/mineflayer-pathfinder#242]
Bot sometimes gets stuck pathfinding on chests.
After hitting the bot in the right direction it fires the goal_reached event and completes.

```
goal_updated
path_update {
  status: 'success',
  cost: 1,
  time: 0.1872999998740852,
  visitedNodes: 1,
  generatedNodes: 5,
  path: [
    Move {
      x: -7855.515625,
      y: 65.875,
      z: 21658.5,
      remainingBlocks: 0,
      cost: 1,
      toBreak: [],
      toPlace: [],
      parkour: false,
      hash: '-7856,66,21658'
    }
  ],
  context: AStar {
    startTime: 138876.9816999999,
    movements: Movements {
      bot: [EventEmitter],
      canDig: false,
      digCost: 1,
      placeCost: 1,
      liquidCost: 1,
      dontCreateFlow: true,
      allow1by1towers: true,
      allowFreeMotion: false,
      allowParkour: true,
      allowSprinting: true,
      blocksCantBreak: [Set],
      blocksToAvoid: [Set],
      liquids: [Set],
      climbables: [Set],
      replaceables: [Set],
      scafoldingBlocks: [Array],
      fences: [Set],
      carpets: [Set],
      maxDropDown: 4
    },
    goal: GoalNear { x: -7854, y: 67, z: 21655, rangeSq: 16 },
    timeout: 5000,
    tickTimeout: 40,
    closedDataSet: Set(1) { '-7856,66,21659' },
    openHeap: BinaryHeapOpenSet { heap: [Array] },
    openDataMap: Map(5) {
      '-7857,67,21659' => [PathNode],
      '-7856,66,21658' => [PathNode],
      '-7856,66,21660' => [PathNode],
      '-7857,66,21658' => [PathNode],
      '-7857,66,21660' => [PathNode]
    },
    bestNode: PathNode {
      data: [Move],
      g: 1,
      h: 4.82842712474619,
      f: 5.82842712474619,
      parent: [PathNode]
    },
    maxCost: -1,
    visitedChunks: Set(1) { '-491,1353' }
  }
}
<=== Infinitely stuck here
goal_reached <== after hitting the bot manually, increasing the Y value
```

**Comment by IceTank** [comment]
What version off mc and what brand off server where you using? Can you provide more detail off the world the bot was walking in? Like was it a double chest or a normal chest? Dose it only get stuck sometimes or every time? Dose this also happen with other blocks that are not full blocks like for instance enchantment tables?

**Comment by TRCYX** [comment]
I also find this happening on soul sand. In here:
https://github.com/PrismarineJS/mineflayer-pathfinder/blob/95369c60bdc32d83c196644e5dd02ecb2d48ed3f/index.js#L513-L520
The position `p` gets floored before comparing to the goal. I believe that A* thinks that standing on the soul sand / chest is already reaching the goal, while if `p` is floored, the distance towards the goal gets a little bit too far. I suppose `floored` is called because the bot could be in mid-air? It is also called in some other occasions, not only this one.

**Comment by TRCYX** [comment]
This happens most of the time in my nether wart farm :(
The bot just gets stuck, no event emits since part of it thinks it reaches the goal while another part thinks it does not.

**Comment by TRCYX** [comment]
Seems that the bug is really old. The position was floored upon the check was added.

**Comment by IceTank** [comment]
> The position `p` gets floored before comparing to the goal.

This is because Goal.isEnd() only takes in integer vectors. Giving vectors to isEnd that are not rounded to the nearest integer would result in a lot off goals breaking.

I can replicate the behavior off the bot getting stuck on blocks that are not full blocks. But I am not sure how this could be fixed.

**Comment by IceTank** [comment]
Potential fix #268

**Comment by IceTank** [comment]
This should be fixed now in this pr #268. Feel free to re open when you find any more issues with this.
